### ISSUE TITLE
Process additional tags for analysis

### ISSUE THREAD

**Issue by nrenner** [abrensch/brouter#249]
We have a new feature in BRouter-Web that shows length distribution for the highway, surface, and smoothness tags ([brouter-web#304](https://github.com/nrenner/brouter-web/pull/304)).

With `processUnusedTags` set to `false` by default, this analysis doesn't get all the tags and values it would need though.

I see following options:

1. Add `processUnusedTags=true` to all profiles (this is what mjaschen does at his instance)
2. Pass a profile parameter from the client that overrides the default `processUnusedTags` setting
3. Pass a list of additional tags to include in the `RoutingContext`  
   (probably hard-coded in the `ServerHandler`)
4. Re-requesting the route when opening the analysis tab with `processUnusedTags=true` as profile parameter or in a modified custom profile
5. Re-requesting the route like in 4., but only when the user presses a button in the analysis tab

I don't know how much always using all tags will degrade performance and memory usage vs. how much issuing additional route requests for all tags on demand will increase overall load. Option 3 might be a compromise.

**Comment by poutnikl** [comment]
I remember Arndt explicitly disabled  processing all tags by default in some BRouter version due performance reasons. I suppose enabling it by default should be limited for   debugging or profile development/testing only.

**Comment by nrenner** [comment]
> But if it's just the smoothness tag missing, why not adding a dummy-usage?
> assign dummyUsage = smoothness=

@mjaschen What do you think about this suggestion? Would you want to make a PR adding such a dummy usage for missing tags to the profiles?

**Comment by mjaschen** [comment]
> What do you think about this suggestion? Would you want to make a PR adding such a dummy usage for missing tags to the profiles?

That's totally fine with me!

For future-proofing the solution: Would it be possible to provide multiple tags/arguments to the `dummyUsage` parameter?

Currently there's only need for the `smoothness=` argument but who knows what surprises the future holds for us … ;-)

**Comment by nrenner** [comment]
> For future-proofing the solution: Would it be possible to provide multiple tags/arguments to the dummyUsage parameter?

Yes, as I understand, `dummyUsage` is just an arbitrary variable name that isn't used anywhere. The idea is to add any valid syntax with no semantic meaning, just to make use of those extra tag(s) in the profile.

Tag names (keys) are [marked as used](https://github.com/abrensch/brouter/blob/0841216cda47d3327b039e6db00c253e86d0948d/brouter-expressions/src/main/java/btools/expressions/BExpression.java#L166) at profile parse time, so multiple tags could for example be added as `or` expressions or as separate assignments to some `dummy1`, `dummy2` variables.

Not sure where to place this in the profiles, probably at the end of the way context, just above the node context.

**Comment by EssBee59** [comment]
@mjaschen
"but who knows what surprises the future holds for us … ;-)!
 Good  remark!
As example, smoothness itself is allready used with further tags in OSM..
(See my request #241)
